include/image-commands.mk

```diff
@@ -316,6 +316,14 @@ define Build/gzip
 	@mv $@.new $@
 endef
 
+define Build/gzip-filename
+	@mkdir -p $@.tmp
+	@cp $@ $@.tmp/$(word 1,$(1))
+	$(STAGING_DIR_HOST)/bin/gzip -f -9 -N -c $@.tmp/$(word 1,$(1)) $(word 2,$(1)) > $@.new
```

Comment:
The `-N` option adds a timestamp. Could you make it reproducible for example by changing the timestamp of `$@.tmp/$(word 1,$(1))` to `$(SOURCE_DATE_EPOCH)`

Reply:
good idea, I'll have to use touch since I'm pretty sure gzip doesn't support custom timestamp change

Reply:
looks good to me.